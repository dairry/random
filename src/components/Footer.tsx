import React, { memo } from 'react';
import { createPortal } from 'react-dom';

const version =
  process.env.NODE_ENV === 'test' ? '1.0.0' : process.env.npm_package_version;

function Footer() {
  const FooterContent = (
    <span>
      <a
        href="https://github.com/SW999/random-number"
        rel="nofollow"
        title="Github page"
      ></a>{' '}
      | Siarhei Vaitehovich | {`v. ${version}`}
    </span>
  );

  return createPortal(FooterContent, document.getElementById('footer'));
}

export default memo(Footer);
