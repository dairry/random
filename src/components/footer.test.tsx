import React from 'react';
import { render } from '@testing-library/react';
import Footer from '~components/Footer';

describe('Footer component', () => {
  beforeAll(() => {
    let portalRoot = document.getElementById('footer');
    if (!portalRoot) {
      portalRoot = document.createElement('div');
      portalRoot.setAttribute('id', 'footer');
      document.body.appendChild(portalRoot);
    }
  });

  it('Footer in a portal should take a snapshot', () => {
    const { baseElement } = render(<Footer />);

    expect(baseElement).toMatchSnapshot();
  });
});
