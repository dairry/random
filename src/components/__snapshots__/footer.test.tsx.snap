// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Footer in a portal should take a snapshot 1`] = `
<body>
  <div
    id="footer"
  >
    <span>
      <a
        href="https://github.com/SW999/random-number"
        rel="nofollow"
        title="Github page"
      >
        <img
          alt="github"
          height="24"
          src="[object Object]"
          width="24"
        />
      </a>
       
      | Siarhei Vaitehovich | 
      v. 1.0.0
    </span>
  </div>
  <div />
</body>
`;
